Intro/README.md

```diff
@@ -0,0 +1,455 @@
+the more difficult to explain by the intricate relationships between the
+relevant concepts.  This is an attempt at a simple explanation --
+simpler than reading hundreds of pages of documents, but then too, too
+light on detail to be immediately useful.
+
+So what is a TPM?  Well, it's a cryptographic co-processor with special
+features to enable "root of trust measurement" (RTM), remote attestation
+of system state, unlocking of local resources that are kept encrypted
+(e.g., filesystems), and more.  A TPM can do those things, and it can do
+it with rich authentication and authorization policies.
+
+Typically a TPM is a hardware module, a chip, though there are firmware,
+virtual, and simulated TPMs as well, all implemented in software.
+
+To simplify things we'll consider only TPM 2.0.  Also to simplify things
+we'll ignore algorithm agility.
+
+Other parts of this [tutorial](README.md) may cover specific concepts in
+much more detail.
+
+# Core Concepts
+
+Some core concepts in the world of TPMs
+
+ - cryptography:
+    - digest functions
+    - hash extension
+    - HMAC
+    - symmetric encryption
+    - asymmetric encryption and/or signature algorithms
+ - platform configuration registers (PCRs)
+ - cryptographic object naming
+ - immutability of object public areas
+ - key hierarchies
+ - key wrapping
+ - limited resources
+    - tickets
+    - resource management
+ - sessions
+ - authorization
+    - restricted cryptographic keys
+    - policies
+ - other object types
+    - non-volatile (NV) indexes
+ - attestation
+
+We'll assume reader familiarity with cryptography so we need not explain
+it.
+
+Authorization is the most important aspect of a TPM, since that's
+ultimately what it exists for: to authorize a system or application to
+perform certain duties when all the desired conditions allow for it.
+
+TPMs have a very rich set of options for authorization.  It's not just
+[policies](#Policies), but also cryptographic object names used with
+restricted keys to allow access only to applications that also have
+other access.
+
+Where to start?  Let's start with hash extension, which may be the only
+trivial concept in the world of TPMs!
+
+## Important Notes
+
+ - Chained implications.
+
+   It is important to note the TPM ecosystem's critical dependence on
+   TPMs performing authorization as specified.  A lot can be
+   accomplished by chaining cryptographic implications.  Sending a
+   secret to a TPM-using application, with the secret encrypted in that
+   TPM's primary endorsement public key is safe when one knows that
+   public key truly is a legitimate hardware TPM's public key: the TPM
+   will not release the secret to the application unless the recipient
+   has some access specified by the sender.  Complex chains of events
+   can be secure provided one knows that it involves legitimate hardware
+   TPMs that otherwise could not be secured.  Therefore the legitimacy
```

Comment:
What does "legitimate hardware TPMs that otherwise could not be secured" mean?

Reply:
`s/that otherwise could not be secured//`

Reply:
I've removed that whole section.